### PR TITLE
chore: Deprecate button clicks for tests

### LINKED ISSUE #192: deprecate dapp test in dappeteer
Deprecate dappeteer testing dapp in favor of using evaluate method.

> do you thing we could start writing tests without adding code to test dapp like:
> ```typescript
> const resultPromise  = testPage.evaluate(async function() {
>    //prepare data
>    return window.ethereum.sendAsync(
>       {
>         method,
>         params,
>         from: accounts[0],
>       });
> });
> await dappteer.signTypedData()
> const result = await resultPromise
> ```

_Originally posted by @mpetrunic in https://github.com/ChainSafe/dappeteer/pull/191#discussion_r1027718482_

### PR DESCRIPTION
**Pull request by Tbaut**
**Short description of work done**

### PR Checklist

- [x] I have run linter locally
- [x] I have run unit and integration tests locally
  - [x] Update configuration the newest version (readme and const)
- [x] Rebased to master branch / merged master

### Changes

- _EXAMPLE - Removed basic auth from the controller_

### Issues

Closes #192

### CODE REVIEW

File: test/dapp/index.html [hunk 1]
```diff
@@ -6,22 +6,9 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
Comment (Tbaut) [inline test/dapp/index.html:6]: I kept this empty page voluntarily, so as not to load any remote page, and have to wait for any network requests. I really think it's cleaner than loading any website. Let me know if you have a strong opinion regarding this.

File: src/types.ts [hunk 2]
```diff
@@ -113,3 +113,13 @@ export type Dappeteer = {
...
+
+export enum AddTokenStatus {
+  SUCCESS = "addTokenResultSuccess",
+  FAIL = "addTokenResultFail",
+}
+
+export enum AddNetworkStatus {
+  SUCCESS = "addNetworkResultSuccess",
+  FAIL = "addNetworkResultFail",
+}
```
Comment (BeroBurny) [inline src/types.ts:125]: Nice idea, but think that boolean is simpler solution for it

cc @Lykhoyda @irubido @mpetrunic

File: tsconfig.json [hunk 3]
```diff
@@ -7,7 +7,8 @@
     "typeRoots": ["./node_modules/@types", "./types"],
     "declaration": true,
     "esModuleInterop": true,
-    "outDir": "dist"
+    "outDir": "dist",
+    "resolveJsonModule": true
```
Comment (BeroBurny) [inline tsconfig.json:11]: You can get rid of this as there are no usages of `json` file

**Review by BeroBurny — CHANGES_REQUESTED** [review]
Overall looks good

File: package.json [hunk 4]
```diff
@@ -70,8 +70,9 @@
     "web3": "1.3.4"
   },
   "peerDependencies": {
+    "playwright": ">=1",
     "puppeteer": ">13",
-    "playwright": ">=1"
+    "web3-utils": ">1"
```
[the remark this answers is not in the document]
Comment (Tbaut) [reply package.json]: :100:

File: tsconfig.json (hunk 3 above)
Comment (Tbaut) [reply tsconfig.json]: Nice catch, solved.

File: src/types.ts (hunk 2 above)
Comment (Tbaut) [reply src/types.ts:125]: Agreed, I was just taking what was there though, and didn't change anything but to declare this in the types. Should I go for a boolean? That's a breaking change in the interface then.

Comment (Tbaut) [reply src/types.ts]: ok, I removed these and used plain booleans :)

**Review by BeroBurny — COMMENTED** [review]
Over all looks good, just one question below

File: test/basic.spec.ts [hunk 5]
```diff
@@ -81,27 +103,33 @@ describe("basic interactions", function () {
...
     await metamask.acceptAddToken();
-    await testPage.waitForSelector("#addTokenResultSuccess");
+    const res = await addTokenPromise;
+    expect(res).to.equal(true);
   });
 
   it("should not add network", async () => {
-    await clickElement(testPage, ".add-network-button");
+    const addNetworkPromise = testPage.evaluate(addNetwork);
+    await metamask.page.waitForTimeout(500);
```
[the remark this answers is not in the document]
Comment (BeroBurny) [reply test/basic.spec.ts]: then using a `pause` from `utils` is a better solution than pausing the MetaMask page.

**Review by BeroBurny — APPROVED** [review]
LGTM

File: test/basic.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Tbaut) [reply test/basic.spec.ts]: ah I see what you mean, sounds good, I'll use this in the future if there's a need.

**Pull request merged**